agent/agent.go

```diff
@@ -518,6 +500,15 @@ func (a *Agent) setupTask(agentCtx, setupCtx context.Context, tcInput *taskConte
 	return tc, setupFailed, shouldExit, nil
 }
 
+func (a *Agent) handleSetupError(err error, tc *taskContext, ctx context.Context, shouldSetLogger bool) (*taskContext, bool, bool, error) {
```

Comment:
`setupCtx` should be the first parameter for Go conventions.

I would also order it as `ctx, tc, err, shouldUseFallbackLogger` since we generally pass in `tc` to a lot of functions.